daemon/core.cpp

```diff
@@ -1547,54 +1547,58 @@ void Core::updateShortcutState(XEvent& event, bool& keyReleaseExpected, unsigned
     IdsByShortcut::iterator idsByShortcut = mIdsByShortcut.find(shortcut);
     if (idsByShortcut != mIdsByShortcut.end())
     {
-        Ids &ids = idsByShortcut.value();
-        switch (mMultipleActionsBehaviour)
-        {
-        case MULTIPLE_ACTIONS_BEHAVIOUR_FIRST:
-        {
-            Ids::iterator lastIds = ids.end();
-            for (Ids::iterator idi = ids.begin(); idi != lastIds; ++idi)
-                if (mShortcutAndActionById[*idi].second->call())
-                {
-                    break;
-                }
-        }
-        break;
+        on_shortcut(idsByShortcut.value());
+    }
+}
 
-        case MULTIPLE_ACTIONS_BEHAVIOUR_LAST:
-        {
-            Ids::iterator firstIds = ids.begin();
-            for (Ids::iterator idi = ids.end(); idi != firstIds;)
+void Core::on_shortcut(const Ids& ids) const
+{
+    switch (mMultipleActionsBehaviour)
+    {
+    case MULTIPLE_ACTIONS_BEHAVIOUR_FIRST:
+    {
+        auto lastIds = ids.end();
+        for (auto idi = ids.begin(); idi != lastIds; ++idi)
```

Comment:
Can we use range loop in this?

Reply:
See the `MULTIPLE_ACTIONS_BEHAVIOUR_LAST` case below (reverse iteration). Just thought it being better to understand this way.